public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -487,7 +495,20 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     if (safeInterval > 1) {
       safeInterval = Math.ceil(safeInterval);
     }
-    return Math.max(interval * intervalFactor, minInterval, safeInterval);
+
+    //Calculate adjusted interval based on the current step option
+    let adjustedInterval = Math.max(dynamicInterval * intervalFactor, stepInterval, safeInterval);
```

Comment:
Can there be a situation when stepOption is not min, max or exact? Wondering why we are calculating it when we are always overwriting it?

Reply:
You're right it will never be anything other than min, max or exact. I'll remove that line. I'm used to other programming languages complaining when you don't define the variable with a value first

Reply:
I tried to remove it but it actually complains about not finding `adjustedInterval`. That's why it was defined before the if statement. Do you know if there's a better way to do it?

Reply:
You can just do `let adjustedInterval`.

Reply:
*The let statement declares a block-scoped local variable, optionally initializing it to a value.* So you don't need to initialise it with value and adjustedInterval will be undefined at that moment.

Reply:
The problem is that the function will then return `number | undefined` and some functions that use the value that `adjustInterval()` return don't allow this type. So we either have to change other functions or do something like `let adjustedInterval = safeInterval`

Reply:
`let adjustedInterval = safeInterval` sounds like the correct idea. 🙂

Reply:
Nice 😄